### PR TITLE
Minor fixes for C API

### PR DESCRIPTION
**Pull request by MI3Guy**
Adds C Shared Library to gitignore and added void to some empty parameter functions.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@ bin/wasm2asm
 bin/s2wasm
 bin/wasm-as
 bin/wasm-dis
+lib/libbinaryen-c.so
```
Comment (kripken) [inline .gitignore]: perhaps it could be `lib/*`?

Comment (MI3Guy) [inline .gitignore]: The bin directory is all specified individually so I was did the same thing. I'll defer to those who know the code better than I do.

Comment (kripken) [inline .gitignore]: Hmm, probably we should change that too, I suppose. Anyhow, doesn't need to be in this PR.

**Comment by kripken** [comment]
Before we merge this, have you joined the [w3c group](https://www.w3.org/community/webassembly/participants)?

**Comment by MI3Guy** [comment]
Ah, Sorry. I just submitted a request to join.

**Comment by kripken** [comment]
Great, thanks!

**Pull request merged**